gph/src/ripser.h

```diff
@@ -691,9 +691,34 @@ class ripser
         };
 
         const index_t cnt = n - (k - 1);
+
         if (pred(n) || (cnt <= 0))
             return n;
-        return get_max(n, cnt, pred);
+
+        if (k != 3) {
+            return get_max(n, cnt, pred);
+        } else {
+            double to_cbrt = 6 * idx;
+            index_t guess = static_cast<index_t>(std::floor(std::cbrt(to_cbrt))) + 1;
+
+            const bool increment = pred(guess);
+
+            while (true) {
+                if (increment) {
+                    ++guess;
+                    if (!pred(guess)) {
+                        --guess;
+                        break;
+                    }
+                } else {
+                    --guess;
+                    if (pred(guess))
+                        break;
+                }
+            }
```

Comment:
We need to give more careful thought about edge cases: for example we mustn't increment if `guess` equals `n - 1` or else we go out of bounds, and we can't decrement if `guess` equals 0.  Maybe this is ensured by `pred` already?